name: bolt-test

on: workflow_dispatch

concurrency:
  group: check-${{ github.ref }}
  cancel-in-progress: true

defaults:
  run:
    # Enable fail-fast behavior
    shell: bash -e {0}

env:
  store: /home/runner/nix

jobs:
  test:
    runs-on: [self-hosted, linux, x64]

    steps:
      # MARK: Git
      - name: Checkout
        uses: actions/checkout@v4
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          submodules: recursive

      - name: LFS
        run: git lfs pull

      - uses: ./.github/actions/pre-init
        with:
          SCCACHE_AWS_SECRET_ACCESS_KEY: ${{ secrets.SCCACHE_AWS_SECRET_ACCESS_KEY }}
          SCCACHE_AWS_ACCESS_KEY_ID: ${{ secrets.SCCACHE_AWS_ACCESS_KEY_ID }}
          OP_SERVICE_ACCOUNT_TOKEN: ${{ secrets.OP_SERVICE_ACCOUNT_TOKEN }}

      - name: Bolt Init
        run: nix-shell --pure --run "bolt init --yes ci"

      - name: Bolt Test
        run: nix-shell --pure --run "bolt test -c 4 -t 120"

      - name: Force Parallel Failure
        if: failure()
        uses: andymckay/cancel-action@0.3
